descqa/BiasVersusRedshift.py

```diff
@@ -0,0 +1,137 @@
+        ax[0].set_title('{} vs. {}'.format(catalog_name, self.data_label), fontsize='medium')
+        ax[1].plot(z,bias)
+        ax[1].set_title('Bias vs redshift', fontsize='medium')
+        ax[1].set_xlabel('$z$')
+        ax[1].set_ylabel('$b(z)$')
+        fig.tight_layout()
+        fig.savefig(os.path.join(output_dir, '{:s}.png'.format(self.test_name)), bbox_inches='tight')
+        plt.close(fig)
+
+    
+    def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
+        '''
+        Loop over magnitude cuts and make plots
+        '''
+        catalog_data = self.load_catalog_data(catalog_instance=catalog_instance,
+                                              requested_columns=self.requested_columns,
+                                              test_samples=self.test_samples)
+
+        if not catalog_data:
+            return TestResult(skipped=True, summary='Missing requested quantities')
+
+        # Initialize catalog's cosmology
+        cosmo = ccl.Cosmology(Omega_c=catalog_instance.cosmology.Om0-catalog_instance.cosmology.Ob0,
+                              Omega_b=catalog_instance.cosmology.Ob0,
+                              h=catalog_instance.cosmology.h,
+                              sigma8=0.8, # For now let's assume a value for 0.8
+                              n_s=0.96 #We assume this value for the scalar index
+                              )
+       
+        rand_cat, rr = self.generate_processed_randoms(catalog_data)
+        
+        correlation_data = dict()
+        nz_data = dict()
+        correlation_theory = dict()
+        best_fit_bias = []
+        z_mean = []
+        for sample_name, sample_conditions in self.test_samples.items():
+            tmp_catalog_data = self.create_test_sample(
+                catalog_data, sample_conditions)
+            with open(os.path.join(output_dir, 'galaxy_count.dat'), 'a') as f:
+                f.write('{} {}\n'.format(sample_name, len(tmp_catalog_data['ra'])))
+            if not len(tmp_catalog_data['ra']):
+                continue
+            z_mean.append(np.mean(tmp_catalog_data['redshift']))
+            output_treecorr_filepath = os.path.join(output_dir, 
+                self.output_filename_template.format(sample_name))
+           
+            xi_rad, xi, xi_sig = self.run_treecorr(
+                catalog_data=tmp_catalog_data,
+                treecorr_rand_cat=rand_cat,
+                rr=rr,
+                output_file_name=output_treecorr_filepath)
+            
+            correlation_data[sample_name] = (xi_rad, xi, xi_sig)
+            nz, be = np.histogram(tmp_catalog_data['redshift'], range=(0,2), bins=100)
+            zcent = 0.5*(be[1:]+be[:-1])
+            nz_data[sample_name] = (zcent, nz*1.0)
+            
+            # Generate CCL tracer object to compute Cls -> w(theta)
+            tracer = ccl.ClTracerNumberCounts(cosmo, has_rsd=False, has_magnification=False,
```

Comment:
`ccl.ClTracerNumberCounts` is no longer available in CCL.

Reply:
Thanks @yymao! I'll update this tomorrow.